runtime/plugin/nvim/net.lua

```diff
@@ -0,0 +1,43 @@
+vim.g.loaded_remote_file_loader = true
+
+--- Callback for BufReadCmd on remote URLs.
+--- @param args { buf: integer }
+local function on_remote_read(args)
+  if vim.fn.executable('curl') ~= 1 then
+    vim.api.nvim_echo({
+      { 'Warning: `curl` not found; remote URL loading disabled.', 'WarningMsg' },
+    }, true, {})
+    return true
+  end
+
+  local bufnr = args.buf
+  local url = vim.api.nvim_buf_get_name(bufnr)
+  local view = vim.fn.winsaveview()
```

Comment:
Is this because it's async? We can try this. But maybe it makes sense to force it to be synchronous by using `vim.wait()`.